develop-docs/sdk/telemetry/logs.mdx

```diff
@@ -0,0 +1,381 @@
+| 9-12                 | Info       | An informational event. Indicates that an event happened.                               |
+| 13-16                | Warn       | A warning event. Not an error but is likely more important than an informational event. |
+| 17-20                | Error      | An error event. Something went wrong.                                                   |
+| 21-24                | Fatal      | A fatal error such as application or system crash.                                      |
+
+Default SDK public API should set the lowest severity number for a given severity level. For example, `warn` severity level logs collected by the SDK API should use the severity number `13`.
+
+## Public API
+
+API wise the SDKs are required to expose logging methods which are to be defined in a `logger` module or namespace. The SDKs should also include some initialization options to configure the behavior of logs in the SDK.
+
+### Initialization Options
+
+The SDKs need to expose the following two configuration options:
+
+- `enableSentryLogs`/`enable_sentry_logs`: A boolean flag to control if log envelopes will be generated and sent to Sentry via the Sentry SDK's Logging APIs. If this flag is set to `false`, the SDK should not send logs to Sentry. Defaults to `false`.
+
+- `beforeSendLog`/`before_send_log`: A function that takes a log object and returns a log object. This function is called before sending the log to Sentry. It can be used to modify the log object or to prevent the log from being sent to Sentry.
+
+```js
+Sentry.init({
+  enableSentryLogs: true,
+
+  beforeSendLog(log) {
+    // Prevent logs from being sent to Sentry if the plan type is enterprise
+    if (log.attributes["plan.type"] === "enterprise") {
+      return null;
+    }
+
+    return log;
+  },
+});
+```
+
+At the current moment, logs are not sampled so no sampling configuration needs to be exposed. This may change in the future.
+
+### Logger Module
+
+At minimum the SDK needs to implement the following two items to be considered to have a complete logging public API:
+
+1. `Sentry.logger.X` (where `X` is the log level): The log levels are `trace`, `debug`, `info`, `warn`, `error`, and `fatal`, which is specced out by the [protocol below](#log-severity-level).
+
+- `Sentry.logger.trace`
+- `Sentry.logger.debug`
+- `Sentry.logger.info`
+- `Sentry.logger.warn`
+- `Sentry.logger.error`
+- `Sentry.logger.fatal`
+
+These methods accepts a string template and the parameters to that string template so the SDK can perform structured logging. Optionally these methods take arbitrary attributes, but not all languages can support both passing a parameterized template and attributes in an easy way.
+
+```js
+// Need to use `fmt` helper function in JavaScript for structured logging.
+Sentry.logger.info(Sentry.logger.fmt`Adding item ${itemId} for user ${userId}`);
+
+Sentry.logger.warn("User %s performed action %s", [userId, actionName], {
+  extra: "123",
+});
```

Comment:
Is it a stupid idea to add overloads for these methods that allow you to pass attributes and severity number as well?  I imagine the following scenario: You have a `warn` log message, but you want to add some attributes to it, which will help you to get more context for fixing a bug. Now you need to refactor `warn` to using `emit(level: warn, ...` which seems a bit annoying.

Reply:
No that is perfectly reasonable. Documented here: ab38e6e88e187f6fbeb59ba88eac5c54b938edab